templates/workspace_services/databricks/terraform/providers.tf

```
@@ -0,0 +1,59 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.9.1"
```

In templates all providers should have a fixed version.
```suggestion
      version = "=3.40.0"
```